### ISSUE TITLE
RoutablePageMixin without WAGTAIL_APPEND_SLASH

### ISSUE THREAD

**Issue by soar** [wagtail/wagtail#2871]
I don't like trailing slashes - they look ugly. When I use `RoutablePageMixin` with `WAGTAIL_APPEND_SLASH = False` template tag `routablepageurl` generates URLs without slash between main part and routable part of URI. For example class:

``` python
class TestPage(RoutablePageMixin, Page):
    @route(r'^tag/(\w+)/$', name='tag')
    def tag_route(self, request, tag, *args, **kwargs):
        # do something
```

For this example on page `example.com/testpage` tag `{% routablepageurl page 'tag' "test" %}` will generate URI `example.com/testpagetag/test` - this wouldn't work of course. I've tried change route to something like `@route(r'^/tag/(\w+)/$', name='tag')` - it throws 404 error.

With `WAGTAIL_APPEND_SLASH = True` - all works nice.

---

And one more problem: if I try route without ending slash:

``` python
class TestPage(RoutablePageMixin, Page):
    @route(r'^tag/(\w+)$', name='tag')
    def tag_route(self, request, tag, *args, **kwargs):
        # do something
```

It doesn't work at all and shows 404 error for both `somepage/tag/test` and `somepage/tag/test/`.

**Comment by coredumperror** [comment]
I would also like to see this fixed. Fortunately, my codebase already had a workaround for the second problem that @soar mentioned (removing the end slash makes the route totally break), so we don't suffer from that. But the templatetag creating the URL wrong means we can't use it, and must build the URL ourselves in a custom templatetag, which is rather ugly:
```
news_index.url + '/' + news_index.reverse_subpage('rss_feed')
```

As you can see by the need for manually inserting a slash, when `WAGTAIL_APPEND_SLASH = False` the reverse_subpage() function doesn't prepend a separator between the route to the Page and the subroutes defined through RoutablePageMixin. 

Perhaps the `wagtailroutablepage.models.route` dectorator could be updated to take `WAGTAIL_APPEND_SLASH = False` into account, and prepend that slash onto the `pattern` argument? If you think that makes sense, I'd be happy to write up a PR.

**Comment by Stormheg** [comment]
It looks like the `routablepageurl` template tag generating incorrect urls was fixed in [PR #3856]( https://github.com/wagtail/wagtail/pull/3856) which was released a while ago in Wagtail 2.0.

Using `@route(r'^tag/(\w+)$', name='tag')` (without a / before the $) still results in 404 errors. In this case it does not matter if `WAGTAIL_APPEND_SLASH` is true or false. I am not sure if this is intended or a bug.

I think this issue can be closed now because the primary issue was resolved.

**Comment by gasman** [comment]
Thanks @Stormheg! Closing this as fixed by #3856, with the secondary issue now covered by #5814.
